x-pack/plugin/inference/src/main/java/org/elasticsearch/xpack/inference/services/cohere/embeddings/CohereEmbeddingsTaskSettings.java

```diff
@@ -106,10 +134,44 @@ public void writeTo(StreamOutput out) throws IOException {
         out.writeOptionalEnum(truncation);
     }
 
+    @Override
+    public boolean equals(Object o) {
+        if (this == o) return true;
+        if (o == null || getClass() != o.getClass()) return false;
+        CohereEmbeddingsTaskSettings that = (CohereEmbeddingsTaskSettings) o;
+        return inputType == that.inputType && truncation == that.truncation;
+    }
+
+    @Override
+    public int hashCode() {
+        return Objects.hash(inputType, truncation);
+    }
+
     public CohereEmbeddingsTaskSettings overrideWith(CohereEmbeddingsTaskSettings requestTaskSettings) {
-        var inputTypeToUse = requestTaskSettings.inputType() == null ? inputType : requestTaskSettings.inputType();
-        var truncationToUse = requestTaskSettings.truncation() == null ? truncation : requestTaskSettings.truncation();
+        if (requestTaskSettings.equals(EMPTY_SETTINGS)) {
+            return this;
+        }
+
+        var inputTypeToUse = requestTaskSettings.getInputType() == null ? inputType : requestTaskSettings.getInputType();
+        var truncationToUse = requestTaskSettings.getTruncation() == null ? truncation : requestTaskSettings.getTruncation();
 
         return new CohereEmbeddingsTaskSettings(inputTypeToUse, truncationToUse);
     }
+
+    /**
+     * Sets the input type field for the task settings if input type value is valid.
+     * @param inputType the new input type to use
+     * @return newly updated task settings
+     */
+    public CohereEmbeddingsTaskSettings setInputType(InputType inputType) {
+        if (VALID_INPUT_VALUES_LIST.contains(inputType) == false) {
```

Comment:
since CohereEmbeddingsTaskSettings is effectively immutable, I would prefer if we used a verb other than `set` for methods like this. Really, I think I would prefer to just add a copy constructor with `InputType` as a second parameter.